samza-core/src/main/java/org/apache/samza/zk/ZkMetadataStore.java

```diff
@@ -18,27 +18,32 @@
  */
 package org.apache.samza.zk;
 
+import com.google.common.primitives.Bytes;
+import com.google.common.primitives.Longs;
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.concurrent.TimeUnit;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.zip.CRC32;
 import org.I0Itec.zkclient.serialize.BytesPushThroughSerializer;
+import org.apache.commons.lang3.ArrayUtils;
 import org.apache.samza.config.Config;
 import org.apache.samza.config.ZkConfig;
 import org.apache.samza.metrics.MetricsRegistry;
 import org.apache.samza.metadatastore.MetadataStore;
 import org.apache.samza.SamzaException;
 import org.I0Itec.zkclient.ZkClient;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
  * An implementation of the {@link MetadataStore} interface where the
  * metadata of the Samza job is stored in zookeeper.
  */
 public class ZkMetadataStore implements MetadataStore {
 
-  private static final Logger LOG = LoggerFactory.getLogger(ZkMetadataStore.class);
+  private static final int VALUE_SEGMENT_SIZE_IN_BYTES = 1020 * 1020;
```

Comment:
Please document why this particular number. It's also better to express it as 1024 * 1024 - XXXX and document that XXXX is headroom for zk stuff.